src/main/res/local_config_winsdk.bzl

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("//tools/cpp:windows_cc_configure.bzl", "find_vc_path", "setup_vc_env_vars")
+load("//tools/cpp:cc_configure.bzl", "MSVC_ENVVARS")
+
+def _find_rc_exe(repository_ctx):
+    vc = find_vc_path(repository_ctx)
+    if vc:
+        env = setup_vc_env_vars(
+            repository_ctx,
+            vc,
+            envvars = [
+                "WindowsSdkDir",
+                "WindowsSdkVerBinPath",
+            ],
+            allow_empty = True,
+            escape = False,
+        )
+
+        # Try the versioned directory.
+        sdk = env.get("WindowsSdkVerBinPath")
+        if sdk:
+            exe = repository_ctx.path(sdk).get_child("x64").get_child("rc.exe")
+            if exe.exists:
+                return str(exe)
+
+        # Try the unversioned directory (typically Windows 8.1 SDK).
+        sdk = env.get("WindowsSdkDir")
+        if sdk:
+            exe = repository_ctx.path(sdk).get_child("bin").get_child("x64").get_child("rc.exe")
+            if exe.exists:
+                return str(exe)
+    return ""
+
+def _impl(repository_ctx):
+    rc_path = _find_rc_exe(repository_ctx)
+
+    repository_ctx.file(
+        "rc_exe.bat",
+        content = ("@echo off\n\"%s\" %%*" % rc_path) if rc_path else ("@echo Could not find Windows SDK.\n@exit /b 1"),
```

Comment:
If feels like you should be able to do something like

rc_path = ...
if not rc_path:
   fail("Could not find Windows SDK")

repository_ctx.file(
￼         "rc_exe.bat",
          ...

But I have no idea if that works

Reply:
That'd be a load-time failure, i.e. you couldn't `bazel query` on Linux.
This is an execution-time failure, i.e. you cannot `bazel build` these rules on Linux.